cmd/list/operatorroles/cmd.go

```diff
@@ -0,0 +1,216 @@
+)
+
+var args struct {
+	clusterKey string
+}
+
+var Cmd = &cobra.Command{
+	Use:     "operator-roles",
+	Aliases: []string{"operator-roles", "operatorroles"},
+	Short:   "List operator roles and policies",
+	Long:    "List operator roles and policies for STS cluster(s)",
+	Example: `  # List operator roles and policies for all clusters
+  rosa list operator-roles
+  # List operator roles with a specific cluster
+  rosa list operator-roles -c mycluster`,
+
+	Run: run,
+}
+
+func init() {
+	flags := Cmd.Flags()
+
+	flags.StringVarP(
+		&args.clusterKey,
+		"cluster",
+		"c",
+		"",
+		"Name or ID of the cluster to list the operator roles for.",
+	)
+	output.AddFlag(Cmd)
+}
+
+func run(_ *cobra.Command, _ []string) {
+	reporter := rprtr.CreateReporterOrExit()
+	logger := logging.CreateLoggerOrExit(reporter)
+
+	// Create the AWS client:
+	awsClient, err := aws.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create AWS client: %v", err)
+		os.Exit(1)
+	}
+
+	awsCreator, err := awsClient.GetCreator()
+	if err != nil {
+		reporter.Errorf("Failed to get AWS creator: %v", err)
+		os.Exit(1)
+	}
+
+	// Create the client for the OCM API:
+	ocmClient, err := ocm.NewClient().
+		Logger(logger).
+		Build()
+	if err != nil {
+		reporter.Errorf("Failed to create OCM connection: %v", err)
+		os.Exit(1)
+	}
+	defer func() {
+		err = ocmClient.Close()
+		if err != nil {
+			reporter.Errorf("Failed to close OCM connection: %v", err)
+		}
+	}()
+
+	// Check that the cluster key (name, identifier or external identifier) given by the user
+	// is reasonably safe so that there is no risk of SQL injection:
+	clusterKey := args.clusterKey
+	var cluster *cmv1.Cluster
+	if clusterKey != "" {
+		if !ocm.IsValidClusterKey(clusterKey) {
+			reporter.Errorf(
+				"Cluster name, identifier or external identifier '%s' isn't valid: it "+
+					"must contain only letters, digits, dashes and underscores",
+				clusterKey,
+			)
+			os.Exit(1)
+		}
+
+		// Try to find the cluster:
+		reporter.Debugf("Loading cluster '%s'", clusterKey)
+		cluster, err = ocmClient.GetClusterOrArchived(clusterKey, awsCreator)
+		if err != nil {
+			fmt.Println(err.Error())
+			reporter.Errorf("Failed to get cluster '%s': %v", clusterKey, err)
+			os.Exit(1)
+		}
+	}
+
+	// get all active clusters
+	activeClusters, err := ocmClient.GetClusters(awsCreator, 1000)
+	if err != nil {
+		reporter.Errorf("Failed to list clusters : '%v'", err)
+		os.Exit(1)
+	}
+
+	// get all archived clusters
+	archivedClusters, err := ocmClient.ListAccountArchivedClusters(awsCreator.AccountID)
+	if err != nil {
+		reporter.Errorf("Failed to list archived clusters : '%v'", err)
+		os.Exit(1)
+	}
+
+	// print message to indicate there may be some wait time
+	totalClusters := len(activeClusters) + len(archivedClusters)
+	if clusterKey != "" {
+		reporter.Infof("Fetching roles for cluster '%s'", clusterKey)
+	} else {
+		reporter.Infof("Fetching roles for %d clusters", totalClusters)
```

Comment:
It dawned on my that this number may not make sense to people... I just ran it and it said `Fetching roles for 86 clusters`, but then the list had only 5 clusters in it. Maybe skip the number or change the message to something like `Fetching roles for %d active clusters and %d archived clusters`, what do you think?